pkg/queue/queue.go

```diff
@@ -9,21 +9,23 @@ import (
 	"time"
 )
 
-func New(initialConcurrency int) *Queue {
+func New(initialConcurrency int, concurrencyPerReplica uint32) *Queue {
 	return &Queue{
-		concurrency: initialConcurrency,
-		list:        list.New(),
-		enqueued:    make(chan struct{}),
-		completed:   make(chan struct{}),
+		concurrency:           initialConcurrency,
+		concurrencyPerReplica: concurrencyPerReplica,
+		list:                  list.New(),
+		enqueued:              make(chan struct{}),
+		completed:             make(chan struct{}),
 	}
 }
 
 // Queue is a thread-safe FIFO queue that will limit the number of concurrent
 // requests that can be in progress at any given time.
 type Queue struct {
-	// concurrency is the max number of in progress items.
-	concurrency    int
 	concurrencyMtx sync.RWMutex
+	// concurrency is the max number of in progress items.
```

Comment:
What is the reason for having both `concurrency` and `concurrencyPerReplica`?

Reply:
@samos123  is correct. Let's have better names. I have updated the doc already:
```
// // concurrency is the max number of in progress items on all replicas. Same as `number_replicas * concurrencyPerReplica`
// concurrencyPerReplica is the max number of in progress items that a single replica instance can handle.
```